backend/director/agents/comparison.py

```
@@ -0,0 +1,138 @@
+                    # Process completed tasks as they finish
+                    completed_count = 0
+                    total = len(futures)
+
+                    while completed_count < total:
+                        res = self.notification_queue.get()
+                        self._update_videos_content(
+                            videos_content, res[0], res[1]
+                        )
+                        completed_count += 1
+
+                    for future in concurrent.futures.as_completed(futures):
```

Ruff (0.8.2)

Loop control variable `future` not used within loop body

(B007)